src/app/ipv/tests/formHandleCoiDetailsCheck.test.ts

```diff
@@ -0,0 +1,226 @@
+import { NextFunction } from "express";
+import { expect } from "chai";
+import sinon from "sinon";
+import proxyquire from "proxyquire";
+
+describe("formHandleCoiDetailsCheck middleware", () => {
+  const testReq = {
+    body: {},
+    params: { pageId: "confirm-your-details" },
+    csrfToken: sinon.fake(),
+    session: {
+      context: "coi",
+      currentPage: "confirm-your-details",
+      save: sinon.fake.yields(null),
+    },
+    log: { error: sinon.fake() },
+  } as any;
+
+  const res = {
+    status: sinon.fake(),
+    redirect: sinon.fake(),
+    send: sinon.fake(),
+    render: sinon.fake(),
+    log: { info: sinon.fake(), error: sinon.fake() },
+    locals: { contactUsUrl: "contactUrl", deleteAccountUrl: "deleteAccount" },
+  } as any;
+
+  let next: NextFunction;
+
+  const coreBackServiceStub = {
+    getProvenIdentityUserDetails: sinon.spy(),
+  };
+
+  const middleware: typeof import("../middleware") = proxyquire(
+    "../middleware",
+    {
+      "../../services/coreBackService": coreBackServiceStub,
+    },
+  );
+
+  beforeEach(() => {
+    next = sinon.fake() as any;
```

Comment:
I can't get `sinon.reset()` working, not sure what I'm missing :/ so for now have had to reassign the variables this way instead to reset any stubs/fakes